src/components/tr/tr.jsx

```diff
@@ -1,160 +1,22 @@
-    }
-
-    const { top: trTop, height } = this.tr.getBoundingClientRect();
-    const offset = this.props.stickyOffset;
-    const tableHeight = this.tr.closest('tbody') ? this.tr.closest('tbody').offsetHeight : this.tr.closest('table').offsetHeight;
-    const fromBottom = this.getDistanceFromBottom() - offset - height;
-    const top = this.state.sticky ? this.tr.nextSibling.getBoundingClientRect().top : trTop;
-
-    if (fromBottom > 0 && top <= offset && top > (tableHeight - height - offset) * -1) {
-      this.setState({
-        sticky: true,
-      });
-    } else if (this.state.sticky) {
-      this.setState({
-        sticky: false,
-      });
-    }
-  }
-
-  handleScroll() {
-    if (this.tr) {
-      this.setSticky(this.tr);
-    }
-  }
-
-  handleResize() {
-    if (this.tr) {
-      this.setWidth(this.tr);
-      this.setSticky(this.tr);
-    }
-  }
-
-  addRef(node, name) {
-    this[name] = node;
-  }
-
-  addClone() {
-    // NOTE: Unfortunately it's impossible to wrap a tr so there's no other way to return two sibling elements.
-    // Returning sibling elements from render will eventually be supported.
-    // https://github.com/facebook/react/issues/2127#issuecomment-232331521
-    const clone = this.tr.cloneNode(true);
-    clone.classList.remove('sticky');
-    clone.classList.add('clone');
-    clone.setAttribute('aria-hidden', true);
-    clone.style.pointerEvents = 'none';
-
-    this.tr.parentNode.insertBefore(clone, this.tr.nextSibling);
-  }
-
-  render() {
-    const { classes, className, children, size, border, variant, stickyOffset, sticky: stickyProp, ...rest } = this.props;
-    const { width, sticky } = this.state;
-    const usedClassName = classNames(
-      classes.tr,
-      size,
-      {
-        [variant]: variant,
-        border,
-        sticky,
-      },
-      className,
-    );
-    const stickyStyle = {};
-
-    if (stickyProp && sticky && width && typeof stickyOffset !== 'undefined') {
-      stickyStyle.top = stickyOffset;
-      stickyStyle.width = width;
-      if (this.tr && this.tr.nextSibling) {
-        this.tr.nextSibling.style.height = 'auto';
-      }
-    } else if (stickyProp && this.tr && this.tr.nextSibling) {
-      this.tr.nextSibling.style.height = 0;
-    }
-
-    return (
-      <tr {...omit(rest, 'sticky', 'theme', 'sheet')} className={usedClassName} style={stickyStyle} ref={node => this.addRef(node, 'tr')}>
-        {children}
-      </tr>
-    );
-  }
+import React from 'react'
+import PropTypes from 'prop-types'
+import injectSheet from 'react-jss'
+import classNames from 'classnames'
+import omit from '../../utilities/omit'
+import styles from './tr-styles'
+
+function Tr({ classes, className, children, size, border, ...rest }) {
+  const usedClassName = classNames(classes.tr, size, { [classes.border]: border }, className)
+  return (
+    <tr {...omit(rest, 'sticky', 'theme', 'sheet')} className={usedClassName}>
```

Comment:
`sticky` is no longer a valid prop. Is this intentional and if so we might want to remove `sticky` from omit

Reply:
fixed